sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -861,9 +857,9 @@ object QueryCompilationErrors {
     new TableAlreadyExistsException(ident)
   }
 
-  def requiresSinglePartNamespaceError(ident: Identifier): Throwable = {
-    new NoSuchTableException(
-      s"V2 session catalog requires a single-part namespace: ${ident.quoted}")
+  def requiresSinglePartNamespaceError(ns: Seq[String]): Throwable = {
+    new AnalysisException(
+      "spark_catalog requires a single-part namespace, but got " + ns.mkString("[", ", ", "]"))
```

Comment:
If you don't mind, can we use `SESSION_CATALOG_NAME` instead of `spark_catalog`?
```scala
- "spark_catalog requires a single-part namespace, but got " + ns.mkString("[", ", ", "]"))
+ s"${SESSION_CATALOG_NAME} requires a single-part namespace, but got " +
+   ns.mkString("[", ", ", "]"))
```